applications/API/tests/test_endpoints.py

```
@@ -0,0 +1,172 @@
+
+class BaseTestCase(TestCase):
+    """
+    Base test case for API endpoint tests.
+
+    This class sets up test users, categories, insults, and an API client for use in endpoint tests.
+    """
+
+    def setUp(self):
+        # Create test users
+        self.user = User.objects.create_user(
+            username="testuser", password="testpass123"
```

Ruff (0.12.2)

Possible hardcoded password assigned to argument: "password"

(S106)